clients/drcachesim/drpt2trace/pt2ir.h

```diff
@@ -0,0 +1,263 @@
+     * redefine some configurations of libipt-sb in sb_config. The libipt-sb sideband
+     * session requires these parameters. We can get these parameters by running
+     * libipt/scirpts/perf-get-opts.bash.
+     */
+    struct {
+        /* perf_event_attr.sample_type */
+        uint64_t sample_type;
+
+        /* The start address of kernel. The sideband session use it to distinguish kernel
+         * from user addresses: kernel >= @kernel_start user   <  @kernel_start
+         */
+        uint64_t kernel_start;
+
+        /* The sysroot is used for remote trace decoding. If the image locates at
+         * /path/to/image in remote machine, it will load it from ${sysroot}/path/to/image
+         * in local machine.
+         */
+        std::string sysroot;
+
+        /* The values of time_shift, time_mult, and time_zero are used to synchronize
+         * trace time, and the sideband recodes time.
+         *
+         *   time_shift = perf_event_mmap_page.ttime_shift
+         *   time_mult = perf_event_mmap_page.time_mult
+         *   time_zero = perf_event_mmap_page.time_zero
+         */
+        uint16_t time_shift;
+        uint32_t time_mult;
+        uint64_t time_zero;
+
+        /* If the trace contains coarse timing information, the sideband events may be
+         * applied too late to cause "no memory mmapped at this address" issue. To set the
+         * tsc_offset can fix the issue.
+         * XXX: The libipt documentation said we need to set a suitable value for
+         * tsc_offset. But it didn't give a method to get a suitable value. So we may need
+         * to find a way to generate the suitable tsc_offset.
+         */
+        uint64_t tsc_offset;
+    } sb_config; /**< The libipt-sb config of PT raw trace. We can get these parameters by
+                  *   running libipt/scirpts/perf-get-opts.bash.
+                  */
+
+    /* sb_primary_file_path is the path of the primary sideband file.
+     * sb_secondary_file_path_list is a list of paths to secondary sideband files.
+     * The sideband primary file and secondary files are stored perf event records. A
+     * primary sideband file contain perf event records for the traced cpu.  A secondary
+     * sideband contain perf event records for other cpus on the system.
+     */
+    std::string sb_primary_file_path; /**< The sideband primary file of PT raw trace. */
+    std::vector<std::string>
+        sb_secondary_file_path_list; /**< The sideband secondary file of PT raw trace. */
+
+    std::string kcore_path; /**< The path of the kernel dump file. */
+};
+
+struct pt_image_section_cache;
+struct pt_sb_pevent_config;
+struct pt_sb_session;
+struct pt_insn_decoder;
+struct pt_ins;
+
+/**
+ * pt2ir_t is a class that can convert PT raw trace to DynamoRIO's IR.
+ */
+class pt2ir_t {
+public:
+    pt2ir_t();
+    ~pt2ir_t();
+
+    /**
+     * Returns TRUE if the pt2ir_t is successfully initialized. Returns FALSE on failure.
```

Comment:
I would not capitalize TRUE and FALSE: those are the names of Windows macros for 32-bit-sized BOOL types.  Use the keyword syntax for C++ one-byte bools.

Reply:
Done.